conan/tools/build/flags.py

```diff
@@ -0,0 +1,457 @@
+from conan.tools.apple.apple import to_apple_arch
+from conans.model.version import Version
+
+
+def architecture_flag(settings):
```

Comment:
Just to make sure, most of these are internal in 2.0 to the toolchains and other helpers. If they are backported to 1.X we should make sure:

- They are not documented as public, they are not exported via ``from conan.tools.xxxx``
- I would expect the consumers (toolchains) using these to have been changed.
- Recall that it is possible that we want to pass ``conanfile`` as the first argument always to all public tools, so this might need a refactor for that.